presto-main/src/test/java/com/facebook/presto/type/TestDecimalOperators.java

```diff
@@ -72,7 +72,7 @@ public void testAdd()
         assertInvalidFunction("DECIMAL '-99999999999999999999999999999999999999' + DECIMAL '-99999999999999999999999999999999999999'", "DECIMAL result exceeds 38 digits: -199999999999999999999999999999999999998");
 
         // test null
-        assertFunction("NULL + DECIMAL '-2'", createDecimalType(1, 0), null);
+        assertFunction("NULL + DECIMAL '-2'", createDecimalType(2, 0), null);
```

Comment:
I think that for arithmetic operations with UNKNOWN type first parameter type should be the type of the result.